clients/voicechat/nemotron-voicechat-client.py

```
@@ -0,0 +1,1648 @@
+
+
+def _validate_tools(tools):
+    """Validate tools against the OpenAI Realtime API tool spec."""
+    if not isinstance(tools, list):
+        raise ValueError("--tools must be a JSON array of tool objects")
+    for i, tool in enumerate(tools):
+        prefix = f"tools[{i}]"
+        if not isinstance(tool, dict):
+            raise ValueError(f"{prefix}: each tool must be an object")
+        if not isinstance(tool.get("name"), str) or not tool["name"]:
+            raise ValueError(f'{prefix}: "name" must be a non-empty string')
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)